### PR TITLE
read_component_properties parsing results in random assignment of val…

### PR DESCRIPTION
**Pull request by mateiiaona**
read_component_properties parsing results in random assignment of values - issue fix

Signed-off-by: mateii <ioana.matei@microfocus.com>

### CODE REVIEW

File: cs-microfocus-dca/src/main/java/io/cloudslang/content/dca/actions/utils/GetSubscriptionParams.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                                       @Param(value = TRUST_PASSWORD, encrypted = true, description = TRUST_PASSWORD_DESC) final String trustPasswordInp,
+                                       @Param(value = KEYSTORE, description = KEYSTORE_DESC) final String keystoreInp,
+                                       @Param(value = KEYSTORE_PASSWORD, encrypted = true, description = KEYSTORE_PASS_DESC) final String keystorePasswordInp,
+                                       @Param(value = CONNECT_TIMEOUT, description = CONNECT_TIMEOUT_DESC) final String connectTimeout,
+                                       @Param(value = SOCKET_TIMEOUT, description = SOCKET_TIMEOUT_DESC) final String socketTimeout,
+                                       @Param(value = USE_COOKIES, description = USE_COOKIES_DESC) final String useCookies,
+                                       @Param(value = KEEP_ALIVE, description = KEEP_ALIVE_DESC) final String keepAlive,
+                                       @Param(value = QUERY_PARAMS, description = QUERY_PARAMS_DESC) String queryParams) {
+
+
```
Comment (liviu-graur) [inline cs-microfocus-dca/src/main/java/io/cloudslang/content/dca/actions/utils/GetSubscriptionParams.java]: I think should be added default values for trustAllRoots, X509HostnameVerifier, proxyPort

File: cs-microfocus-dca/src/main/java/io/cloudslang/content/dca/actions/utils/GetSubscriptionParams.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            int nr = 0;
+            for (int j = 0; j < arrayProp.length; j++)
+                if (arrayProp[j].contains("param_"))
+                    nr++;
+
+            for (int i = 0; i < arrayProp.length; i++) {
+
+                if (arrayProp[i].contains("param_")) {
+
+                    String xPathQueryChild = "//options/property[name=\"" + arrayProp[i] + "\"]/values/*[name()=\"value\" ]/text()";
```
Comment (liviu-graur) [inline cs-microfocus-dca/src/main/java/io/cloudslang/content/dca/actions/utils/GetSubscriptionParams.java]: Can this be moved into Constants class?

File: cs-microfocus-dca/src/main/java/io/cloudslang/content/dca/actions/utils/GetSubscriptionParams.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            List<String> finalList = new ArrayList<>();
+
+            String[] arrayProp = query_Property_Name.get("selectedValue").split(",");
+
+            int nr = 0;
+            for (int j = 0; j < arrayProp.length; j++)
+                if (arrayProp[j].contains("param_"))
+                    nr++;
+
+            for (int i = 0; i < arrayProp.length; i++) {
```
Comment (ionutmarisca) [inline cs-microfocus-dca/src/main/java/io/cloudslang/content/dca/actions/utils/GetSubscriptionParams.java]: Maybe introduce a new method in a `Service` package

File: cs-microfocus-dca/src/main/java/io/cloudslang/content/dca/actions/utils/GetSubscriptionParams.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                    if (i < nr && nr > 1) {
+                        finalList.add(arrayProp[i] + "=" + query_Child.get("selectedValue") + "&");
+                    } else
+                        finalList.add(arrayProp[i] + "=" + query_Child.get("selectedValue"));
+
+                }
+
+            }
+
+            String finalResponse = "";
```
Comment (ionutmarisca) [inline cs-microfocus-dca/src/main/java/io/cloudslang/content/dca/actions/utils/GetSubscriptionParams.java]: As a suggestion,  you can try to use `StringUtils.join`

**Pull request merged**